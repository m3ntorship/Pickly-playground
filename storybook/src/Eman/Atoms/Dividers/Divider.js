import React from "react";
import PropTypes from "prop-types";

export const DIVIDER = {
  TYPE: {
    HORIZONTAL: "Horizontal",
    VERTICAL: "Vertical",
  },
  RANGE: {
    MIN: "4",
    MAX: "20",
    STEP: "1px",
  },
};

export const Divider = ({ type, range }) => {
  return (
    <>
      {type === DIVIDER.TYPE.HORIZONTAL && (
        <div className={`h-0.5 bg-grey-shd6 w-${range}`}></div>
      )}
      {type === DIVIDER.TYPE.VERTICAL && (
        <div className={`w-0.5 bg-grey-shd6 h-${range}`}></div>
      )}
    </>
  );
};

Divider.propTypes = {
  type: PropTypes.string,
  range: PropTypes.any,
};
