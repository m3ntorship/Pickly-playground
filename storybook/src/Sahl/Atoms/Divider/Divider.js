import React from "react";
import PropTypes from "prop-types";

export default function Divider(props) {
  const { className, direction = "horizontal" } = props;
  return (
    // h-32 added only for purpose of demonestration in storybook. Later it should be h-full
    <div
      className={`bg-grey-shd6 ${
        direction === "horizontal" ? "w-full h-px" : "h-32 w-px"
      } ${className}`}
    ></div>
  );
}

Divider.propTypes = {
  direction: PropTypes.oneOf(["horizontal", "vertical"]),
  // className: PropTypes.string,
};
