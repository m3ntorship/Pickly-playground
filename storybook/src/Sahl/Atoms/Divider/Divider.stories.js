import React from "react";
import Divider from "./Divider";

const defaultExport = {
  title: "Sahl/Atoms/Divider",
  component: Divider,
};

export default defaultExport;

const Template = (args) => <Divider {...args} />;

export const Default = Template.bind({});
Default.args = {
  direction: "horizontal",
};
