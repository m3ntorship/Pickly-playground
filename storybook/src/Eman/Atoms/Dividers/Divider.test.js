import React from "react";
import renderer from "react-test-renderer";
import { Divider } from "./Divider";

describe("Testing Divider", () => {
  it("Testing Checkbox type Horizontal", () => {
    const tree = renderer.create(<Divider type="Horizontal" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing Divider type Vertical", () => {
    const tree = renderer.create(<Divider type="Vertical" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing Divider with no props", () => {
    const tree = renderer.create(<Divider />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
