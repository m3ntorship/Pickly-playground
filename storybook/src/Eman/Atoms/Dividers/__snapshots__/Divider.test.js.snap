// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Divider Testing Checkbox type Horizontal 1`] = `
<div
  className="h-0.5 w-undefined bg-grey-shd6"
  style={
    Object {
      "backgroundColor": "",
    }
  }
/>
`;

exports[`Testing Divider Testing Divider type Vertical 1`] = `
<div
  className="w-0.5 h-undefined bg-grey-shd6"
  style={
    Object {
      "backgroundColor": "",
    }
  }
/>
`;

exports[`Testing Divider Testing Divider with no props 1`] = `null`;
